Stack/Opc.Ua.Core/Types/Encoders/BinaryEncoder.cs

```diff
@@ -1917,6 +1926,30 @@ public void WriteArray(string fieldName, object array, int valueRank, BuiltInTyp
         #endregion
 
         #region Private Methods
+        /// <summary>
+        /// Writes a byte string to the stream from a given index and length.
+        /// </summary>
+        private void WriteByteString(string fieldName, byte[] value, int index, int count)
+        {
```

Comment:
may be useful to have this API signature available in IEncoder 
Also looking into ReadOnlySpan<byte>